### PR TITLE
:bug: Fix for exit menu disappearing

### PR DESCRIPTION
**Pull request by rhyfii**
**Describe Pull request:**
Fix for exit menu disappearing on invited player relog. Ensures exit target is not deleted on DeleteHousesTargets() function

**If your PR is to fix an issue mention that issue here:**
When first player entered instance and invited another player and the invited player relogged, when the invited player selected to respawn back within the instance, the exit PolyZone would be destroyed 

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [yes/no] (Be honest) Yes
- Does your code fit the style guidelines? [yes/no] Yes
- Does your PR fit the contribution guidelines? [yes/no] Yes

### CODE REVIEW

**Comment by jnccloud** [comment]
I'm putting this PR into production tomorrow morning. I've tested in dev by going AFK for 25 mins. Seemed to fix the issue at hand. Will update in a few days the results; feel free to tag me if I've not responded in 48 hours.

**Comment by nzkfc** [comment]
> 

Hows it been going?

**Comment by jnccloud** [comment]
> > 
> 
> Hows it been going?

It worked perfectly for me. So good, I forgot I put it in. :) 32 days no issues with it

**Comment by nzkfc** [comment]
@GhzGarage So my server owner wants to replace qb-houses and qb-garages in full because it's created a bit of a headache for players being stuck inside. You mentioned you wanted me to recreate the issue and after trying to do so on a dev city, I couldn't so just before restart tonight I enabled debugs on qb-garages and qb-houses and went to the worst house we have for this issue.

Ignore the random sounds in the video, it's a new script being tested.

https://www.youtube.com/watch?v=rOTJ5MJYCEA

**Things to note:**

- When I drive into the polyzone for the garage, it's not there initially, it loads as I pass it and can then use it. Previously this box the garage was overlapping the house polyzone but we moved it, but issue still exists.
- Basically looks like when you go into the house, the exit poly goes poof after a few seconds or if you move away from it, seems to be intermittent but you can actually see it happening.
- I've compared our qb-houses, qb-interior and qb-garages to a fresh pull and it's all identical except for a few config settings/XYZ etc.

Pretty sure it's just down to the various bug reports and PR's sitting open on qb-houses, qb-garages that need going through as you mentioned on stream, that the dev team got thinned out but just need time/resource to get stuck into it.

Anyways let me know if you want my version of the script, happy to even send our DB so you can load that into a city and play around with it etc, just let me know.

**Comment by nzkfc** [comment]
Hmm this pops up when I restart the script, not sure if related:
https://i.imgur.com/9SQlMiB.jpeg

**Comment by jnccloud** [comment]
> Hmm this pops up when I restart the script, not sure if related: https://i.imgur.com/9SQlMiB.jpeg

I've fixed most of these issues in my version of the script. I will see about doing a PR. It also adds a few new features like house management, admin creation commands, etc.

**Comment by jnccloud** [comment]
> @GhzGarage So my server owner wants to replace qb-houses and qb-garages in full because it's created a bit of a headache for players being stuck inside. You mentioned you wanted me to recreate the issue and after trying to do so on a dev city, I couldn't so just before restart tonight, on a main city with 100 odd players at the time, I enabled debugs on qb-garages and qb-houses and went to the worst house we have for this issue.
> 
> Ignore the random sounds in the video, it's a new script being tested.
> 
> https://www.youtube.com/watch?v=rOTJ5MJYCEA
> 
> **Things to note:**
> 
> * When I drive into the polyzone for the garage, it's not there initially, it loads as I pass it and can then use it. Previously this box the garage was overlapping the house polyzone but we moved it, but issue still exists.
> * Basically looks like when you go into the house, the exit poly goes poof after a few seconds or if you move away from it, seems to be intermittent but you can actually see it happening.
> * I've compared our qb-houses, qb-interior and qb-garages to a fresh pull and it's all identical except for a few config settings/XYZ etc.
> 
> Pretty sure it's just down to the various bug reports and PR's sitting open on qb-houses, qb-garages that need going through as you mentioned on stream, that the dev team got thinned out but just need time/resource to get stuck into it.
> 
> Anyways let me know if you want my version of the script, happy to even send our DB so you can load that into a city and play around with it etc, just let me know.

Curious - how did you enable debug during a live session, did you restart the script? If you do, qb-houses doesn't play nice with a hot reload.

**Comment by nzkfc** [comment]
Yep restarted the script, worked fine for us.

Got one player today saying he couldn't exit so will look into that tonight
since implementing your change.

**Comment by nzkfc** [comment]
Can confirm been using this for 16 days and it's working fine.
